Tsc in watch mode does not seem to observe changes when compiler errors due to project references are fixed

[TscBugOutdatedErrors.zip](https://github.com/Microsoft/TypeScript/files/2465160/TscBugOutdatedErrors.zip)

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  3.2.0-dev.20181010

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms: watch project reference **

Please find attached a folder with some files to reproduce the Problem.
The folder contains two subfolders:
1) A folder 'library' with an typescript file and a simple tsconfig.json
2) A folder 'app' with an typescript file and a tsconfig.json, referencing the library Project.

For observing the bug, please open a command prompt and navigate to the root Folder.
1) From command line, run "tsc -v --watch App" to transpile the files. Initially, there are no errors.
2) Edit Library\Library.ts: Rename the property 'message" of 'SomeObject' to 'message1' in the interface declaration as well as in the method 'createSomeObject()'. Save the file. The compiler reports an error in App/app.ts stating that the property 'message' cannot be found. This is correct.
3) Undo the changes from step 2, i.e. replace 'message1' by 'message' in both occurences and save the file. 
**Expected behavior:**
Now I expect no errors. 

**Actual behavior:**
However, the error message from step 2 still remains, as if the compiler did not notice the changes.

If I kill the watch process and restart the compiler again, everything is fine.

@ThomasKiesel are you using tsc -b or tsc -p to build the project?  I tried with tsc -b since you used -v flag I assumed that what you want.. I cant repro this at all. I was able to see the addition of error when I renamed library message, and removal when I did undo of the change.

Also note that with both of these I am getting errors:
`
pp/app.ts:5:1 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

5 console.log("SomeObject says: " + createSomeObject().message);
  ~~~~~~~

App/app.ts:6:1 - error TS2304: Cannot find name 'setTimeout'.

6 setTimeout(function (){}, 3000);
  ~~~~~~~~~~
`

Can you copy paste the logs from --v that you get. May be that can give idea as to what might be going wrong.

[TscBugOutdatedErrors.zip](https://github.com/Microsoft/TypeScript/files/2465810/TscBugOutdatedErrors.zip)
I have to apologize for submitting insufficient information. I had some trouble with sending the zip-archive and reduced the content to upload it, obviously it was not enough. 

Please find attached the full zip file. Here you find an additional folder “node_modules”, containing the node type definitions and the typescript compiler which I used. 

I run the command “node_modules\.bin\tsc -b --watch App” in the root folder. “App” is the target project to compile. 
After step 2, the output is as follows:
```

_App/app.ts:5:54 - error TS2551: Property 'message' does not exist on type 'SomeObject'. Did you mean 'message1'?

5 console.log("SomeObject says: " + createSomeObject().message);
                                                       ~~~~~~~

  Library/library.d.ts:2:5
    2     message1: string;
          ~~~~~~~~
    'message1' is declared here.

[19:41:18] Found 1 error. Watching for file changes._

```
After step 3, these messages disappear for a moment, just to return again:
```
__App/app.ts:5:54 - error TS2551: Property 'message' does not exist on type 'SomeObject'. Did you mean 'message1'?

5 console.log("SomeObject says: " + createSomeObject().message);
                                                       ~~~~~~~

  Library/library.d.ts:2:5
    2     message1: string;
          ~~~~~~~~
    'message1' is declared here._

[19:42:45] Found 1 error. Watching for file changes._`
```

--- a/src/compiler/tsbuild.ts
+++ b/src/compiler/tsbuild.ts
@@ -1066,6 +1066,9 @@ namespace ts {
                 type: UpToDateStatusType.UpToDate,
                 newestDeclarationFileContentChangedTime: anyDtsChanged ? maximumDate : newestDeclarationFileContentChangedTime
             };
+            if (options.watch) {
+                diagnostics.removeKey(proj);
+            }
             projectStatus.setValue(proj, status);
             return resultFlags;
 

--- a/src/testRunner/unittests/tsbuildWatchMode.ts
+++ b/src/testRunner/unittests/tsbuildWatchMode.ts
@@ -356,6 +356,58 @@ function myFunc() { return 100; }`);
             }
         });
 
+        it("when referenced project change introduces error in the down stream project and then fixes it", () => {
+            const subProjectLibrary = `${projectsLocation}/${project}/Library`;
+            const libraryTs: File = {
+                path: `${subProjectLibrary}/library.ts`,
+                content: `
+interface SomeObject
+{
+	message: string;
+}
+
+export function createSomeObject(): SomeObject
+{
+	return {
+		message: "new Object"
+	};
+}`
+            };
+            const libraryTsconfig: File = {
+                path: `${subProjectLibrary}/tsconfig.json`,
+                content: JSON.stringify({ compilerOptions: { composite: true } })
+            };
+            const subProjectApp = `${projectsLocation}/${project}/App`;
+            const appTs: File = {
+                path: `${subProjectApp}/app.ts`,
+                content: `import { createSomeObject } from "../Library/library";
+createSomeObject().message;`
+            };
+            const appTsconfig: File = {
+                path: `${subProjectApp}/tsconfig.json`,
+                content: JSON.stringify({ references: [{ path: "../Library" }] })
+            };
+
+            const files = [libFile, libraryTs, libraryTsconfig, appTs, appTsconfig];
+            const host = createWatchedSystem(files, { currentDirectory: `${projectsLocation}/${project}` });
+            createSolutionBuilderWithWatch(host, ["App"]);
+            checkOutputErrorsInitial(host, emptyArray);
+
+            // Change message in library to message2
+            host.writeFile(libraryTs.path, libraryTs.content.replace(/message/g, "message2"));
+            host.checkTimeoutQueueLengthAndRun(1); // Build library
+            host.checkTimeoutQueueLengthAndRun(1); // Build App
+            checkOutputErrorsIncremental(host, [
+                "App/app.ts(2,20): error TS2551: Property 'message' does not exist on type 'SomeObject'. Did you mean 'message2'?\n"
+            ]);
+
+            // Revert library changes
+            host.writeFile(libraryTs.path, libraryTs.content);
+            host.checkTimeoutQueueLengthAndRun(1); // Build library
+            host.checkTimeoutQueueLengthAndRun(1); // Build App
+            checkOutputErrorsIncremental(host, emptyArray);
+        });
+
         describe("reports errors in all projects on incremental compile", () => {
             function verifyIncrementalErrors(defaultBuildOptions?: BuildOptions, disabledConsoleClear?: boolean) {
                 const host = createSolutionInWatchMode(allFiles, defaultBuildOptions, disabledConsoleClear);

